network/setup-ip

```diff
@@ -6,20 +6,24 @@
 
 have_qubesdb || exit 0
 
-ip=$(/usr/bin/qubesdb-read /qubes-ip 2> /dev/null)
-ip6=$(/usr/bin/qubesdb-read /qubes-ip6 2> /dev/null)
-if [ "x$ip" != x ]; then
-    #netmask=$(/usr/bin/qubesdb-read /qubes-netmask)
-    gateway=$(/usr/bin/qubesdb-read /qubes-gateway)
-    gateway6=$(/usr/bin/qubesdb-read /qubes-gateway6)
-    primary_dns=$(/usr/bin/qubesdb-read /qubes-primary-dns 2>/dev/null || echo "$gateway")
-    secondary_dns=$(/usr/bin/qubesdb-read /qubes-secondary-dns)
-    /sbin/ethtool -K "$INTERFACE" sg off
-    /sbin/ethtool -K "$INTERFACE" tx off
-    # If NetworkManager is enabled, let it configure the network
-    if qsvc network-manager ; then
-        nm_config=/etc/NetworkManager/system-connections/qubes-uplink-$INTERFACE
-        cat > "$nm_config" <<__EOF__
+mac="$(/usr/bin/qubesdb-read /qubes-mac 2> /dev/null)"
+current_mac="$(get_mac_from_iface "$INTERFACE")"
+
+if [ "$mac" = "$current_mac" ]; then
```

Comment:
This breaks the network, if `/qubes-mac` is missing.
Please test how it works without dom0 change in place too.